### PR TITLE
Clarify that clustering option is for shape sources

### PR DESCRIPTION
**Pull request by jmkiley**
Added note that clustering needs a GeoJSON source

### CODE REVIEW

File: platform/darwin/src/MGLShapeSource.h [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #import "MGLSource.h"
 
+#import "MGLFoundation.h"
```
Comment (1ec5) [inline platform/darwin/src/MGLShapeSource.h]: This header is new in the master branch, hence the build failures. In the release-ios-v3.4.0 branch, you don’t need it.

File: platform/darwin/src/MGLShapeSource.h [hunk 2]
```diff
@@ -14,21 +15,23 @@ typedef NSString *MGLShapeSourceOption NS_STRING_ENUM;
 
 /**
  An `NSNumber` object containing a Boolean enabling or disabling clustering.
- If the `shape` property contains point shapes, setting this option to
+ If the `shape` property contains point shapes defined by a GeoJSON source, setting this option to
```
Comment (1ec5) [inline platform/darwin/src/MGLShapeSource.h]: This option can only be placed in a shape source (aka GeoJSON source).

File: platform/darwin/src/MGLShapeSource.h [hunk 3]
```diff
@@ -14,21 +15,23 @@ typedef NSString *MGLShapeSourceOption NS_STRING_ENUM;
...
  `YES` clusters the points by radius into groups. The default value is `NO`.
 
  This attribute corresponds to the
  <a href="https://www.mapbox.com/mapbox-gl-style-spec/#sources-geojson-cluster"><code>cluster</code></a>
  source property in the Mapbox Style Specification.
+ 
+ @note Clustering is currently supported for shapes with a GeoJSON source.
```
Comment (1ec5) [inline platform/darwin/src/MGLShapeSource.h]: This SDK refers to GeoJSON sources as shape sources:

> This option only affects point features within a shape source.

Comment (1ec5) [inline platform/darwin/src/MGLShapeSource.h]: It isn’t necessary to use `@note` here.

File: platform/darwin/src/MGLShapeSource.h [hunk 4]
```diff
@@ -14,21 +15,23 @@ typedef NSString *MGLShapeSourceOption NS_STRING_ENUM;
...
  */
-extern const MGLShapeSourceOption MGLShapeSourceOptionClustered;
+extern MGL_EXPORT const MGLShapeSourceOption MGLShapeSourceOptionClustered;
```
Comment (1ec5) [inline platform/darwin/src/MGLShapeSource.h]: `MGL_EXPORT` is also new to master and unnecessary on this branch.

File: platform/darwin/src/MGLShapeSource.h [hunk 5]
```diff
@@ -107,6 +110,7 @@ extern const MGLShapeSourceOption MGLShapeSourceOptionSimplificationTolerance;
  mapView.style?.addSource(source)
  ```
  */
+MGL_EXPORT
```
Comment (1ec5) [inline platform/darwin/src/MGLShapeSource.h]: `MGL_EXPORT` is new to the master branch and unnecessary on this branch.

**Comment by 1ec5** [comment]
This PR fixes #7821.

File: platform/darwin/src/MGLShapeSource.h [hunk 6]
```diff
@@ -20,6 +20,8 @@ typedef NSString *MGLShapeSourceOption NS_STRING_ENUM;
  This attribute corresponds to the
  <a href="https://www.mapbox.com/mapbox-gl-style-spec/#sources-geojson-cluster"><code>cluster</code></a>
  source property in the Mapbox Style Specification.
+ 
+ This option only affects point features within a shape source..
```
Comment (friedbunny) [inline platform/darwin/src/MGLShapeSource.h]: There are two periods.

**Pull request merged**